app-games/src/main/java/com/aptoide/android/aptoidegames/MainActivity.kt

```diff
@@ -108,7 +111,25 @@ class MainActivity : AppCompatActivity() {
 
   override fun onNewIntent(intent: Intent?) {
     super.onNewIntent(intent)
+    checkDeeplinkContext(intent)
     navController?.handleDeepLink(intent)
     handleNotificationIntent(intent)
   }
+
+  private fun checkDeeplinkContext(intent: Intent?) {
```

Comment:
Why do we want to modify the activity intent instead of returning here the modified intent for further usage? 🤔 
Looks like it currently relies on this function being called only in `onNewIntent` and on the internal details of android regarding relation between intent of the activity and intent received in `onNewIntent`. In other words on implicit platform behaviour...